database.hh

```diff
@@ -870,7 +879,19 @@ public:
     }
 
     bool compaction_enforce_min_threshold() const {
-        return _config.compaction_enforce_min_threshold;
+        return _config.compaction_enforce_min_threshold || _is_bootstrap_or_replace;
+    }
+
+    unsigned min_compaction_threshold() {
```

Comment:
This is kind of wonky. For the purpose of the config variable compaction_enforce_min_threshold, we don't touch the min_compaction_threshold() and set/clear a flag whether to ignore it or not. But for bootstrap we do touch the variable, and set the "ignore" flag to false.

It's not easy to simplify, and now it's going to be even harder.

Reply:
It is not that hard to simplify.
All we have to do is delete this code altogether when offstrategy is ready.

I agree this is a bit weird for now, but after your comment I am not left to guess again what is on your mind.
- Are you against this patch because of that?
- Would you like to pursue a different strategy?
- Are you just saying that is ugly but there's nothing to do?

Reply:
I'm saying it's making things worse but as it's "temporary" we can live with it.